app/controllers/search_controller.rb

```diff
@@ -48,7 +52,10 @@ def show
     search_args[:ip_address] = request.remote_ip
     search_args[:user_id] = current_user.id if current_user.present?
 
-    if site_overloaded?
+    if rate_limit_errors
+      result = Search::GroupedSearchResults.new(search_args[:type_filter], @search_term, context, false, 0)
```

Comment:
Ah is the intention for us to return the cached results if rate limit has been exceeded?

Reply:
Oops sorry just read the thread on dev. This looks good to me.